internal/datacoord/compaction.go

```diff
@@ -173,17 +182,11 @@ func (c *compactionPlanHandler) execCompactionPlan(signal *compactionSignal, pla
 			log.Warn("Alloc start time for CompactionPlan failed", zap.Int64("planID", plan.GetPlanID()))
 			return
 		}
-		plan.StartTime = ts
 
 		c.mu.Lock()
-		task := &compactionTask{
-			triggerInfo: signal,
-			plan:        plan,
-			state:       executing,
-			dataNodeID:  nodeID,
-		}
-		c.plans[plan.PlanID] = task
-		c.executingTaskNum++
+		c.plans[plan.PlanID] = c.plans[plan.PlanID].shadowClone(func(task *compactionTask) {
```

Comment:
shadowClone copy only the pointer of plan, does this modification of the original plan wanted?